xla/stream_executor/device_memory.h

```diff
@@ -73,6 +73,11 @@ class DeviceMemoryBase {
   void *opaque() { return opaque_; }
   const void *opaque() const { return opaque_; }
 
+  // This function is called when memory allocation is not ready when the object
+  // is constructed, e.g, the memory allocation node in GPU cuda-graph is
+  // available when the after memory allocation node is added to the graph.
+  void SetOpaque(void *dptr) { opaque_ = dptr; }
```

Comment:
Can you maybe just do an assingment to a pointer instead of opening implementation details:

```
Allocate(se::DeviceMemoryBase* ptr) {
*ptr = se::DeciceMemoryBase(allocated_ptr, allocated_size);
}
```